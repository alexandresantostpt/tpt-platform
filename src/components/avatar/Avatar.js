import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import ReactTooltip from 'react-tooltip'
import If from '../if/If'
import DefaultAvatar from './DefaultAvatar'
import { isNotNull } from '@/utils/functions'

const StyledContainer = styled.div`
    background: url(${({ image }) => image}) no-repeat center;
    background-size: cover;
    border-radius: 7.75rem;
    height: ${({ size }) => size}rem;
    margin-left: 0.3125rem;
    width: ${({ size }) => size}rem;
`

const Avatar = ({ alt, size, image, ...props }) => (
    <>
        <If condition={isNotNull(image)} el={<DefaultAvatar alt={alt} data-tip={alt} size={size} {...props} />}>
            <StyledContainer data-tip={alt} image={image} size={size} {...props} />
        </If>
        <ReactTooltip effect="solid" place="top" type="dark" />
    </>
)

Avatar.defaultProps = {
    image: null,
    onClick: null,
    size: 2.3125
}

Avatar.propTypes = {
    alt: PropTypes.string.isRequired,
    image: PropTypes.string,
    onClick: PropTypes.func,
    size: PropTypes.number
}

export default Avatar
