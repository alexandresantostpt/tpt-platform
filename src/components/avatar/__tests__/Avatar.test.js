import React from 'react'
import renderer from 'react-test-renderer'

import Avatar from '../Avatar'

describe('Tests for Avatar component', () => {
    it('Should render without crash', () => {
        const tree = renderer.create(<Avatar alt="A" />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('Should render with given size', () => {
        const tree = renderer.create(<Avatar alt="A" size={20} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('Should render with given name', () => {
        const tree = renderer.create(<Avatar alt="Harry Potter" />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
