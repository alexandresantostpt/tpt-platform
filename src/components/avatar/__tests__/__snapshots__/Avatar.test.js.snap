// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Avatar component Should render with given name 1`] = `
Array [
  <div
    className="DefaultAvatar__AvatarContainer-sc-1lk30og-0 dWocpW"
    data-tip="Harry Potter"
    onClick={null}
    size={2.3125}
  >
    HP
  </div>,
  <div
    className="__react_component_tooltip place-top type-dark "
    data-id="tooltip"
  />,
]
`;

exports[`Tests for Avatar component Should render with given size 1`] = `
Array [
  <div
    className="DefaultAvatar__AvatarContainer-sc-1lk30og-0 iFCCbd"
    data-tip="A"
    onClick={null}
    size={20}
  >
    A
  </div>,
  <div
    className="__react_component_tooltip place-top type-dark "
    data-id="tooltip"
  />,
]
`;

exports[`Tests for Avatar component Should render without crash 1`] = `
Array [
  <div
    className="DefaultAvatar__AvatarContainer-sc-1lk30og-0 dWocpW"
    data-tip="A"
    onClick={null}
    size={2.3125}
  >
    A
  </div>,
  <div
    className="__react_component_tooltip place-top type-dark "
    data-id="tooltip"
  />,
]
`;
